spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestViews.java

```diff
@@ -213,10 +213,13 @@ public void readFromViewUsingNonExistingTable() throws NoSuchTableException {
 
     assertThatThrownBy(() -> sql("SELECT * FROM %s", viewName))
         .isInstanceOf(AnalysisException.class)
-        .hasMessageContaining(
```

Comment:
I would prefer if we could keep the error msg check. I think it's perfectly fine to keep this and then just add the `.satisfies` check below

Reply:
If we really only care about the error class, then an alternative check could be the one below:

```
assertThatThrownBy(() -> sql("SELECT * FROM %s", viewName))
        .isInstanceOf(AnalysisException.class)
        .hasMessageContaining(
            String.format(
                "The table or view `%s`.`%s`.`non_existing` cannot be found",
                catalogName, NAMESPACE))
        .asInstanceOf(InstanceOfAssertFactories.type(AnalysisException.class))
        .extracting(AnalysisException::getErrorClass)
        .isEqualTo("INVALID_VIEW_TEXT");
```

Reply:
Thanks @nastra for your comment! I'm concerned that checking for specific error messages might make the tests a bit fragile. While this particular error message seems stable, others might change between versions. What do you think?